### DISCUSSION TITLE
Trying to do initial push

### DISCUSSION THREAD

**Discussion by mdiehlenator** [community/community#147556] [Repositories]
I've got other repo's working, but when I went to configure a new repo, I ran into some issues.

We did the code development before worrying about change management.  So, once the (trivial) code was complete, we created a repo on github and tried to push our code to it.

git push --set-upstream origin main
Enumerating objects: 168, done.
Counting objects: 100% (168/168), done.
Delta compression using up to 8 threads
Compressing objects: 100% (161/161), done.
Writing objects: 100% (168/168), 102.92 MiB | 6.12 MiB/s, done.
Total 168 (delta 47), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (47/47), done.
remote: error: Trace: ca0cc1485c7f2aaf6f10aac3c0a128826c88f7e5b255742d0dcac7298f9bd3c9
remote: error: See https://gh.io/lfs for more information.
remote: error: File OLD/orig/ecom_account_116_for_mike_ebay2.csv is 162.75 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/webretailgroup/E-Bay_Crawler.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/webretailgroup/E-Bay_Crawler.git'

I have verified that there is NOTHING in ./OLD/*

What I would like is to be able to issue a git push, and have my local code-base pushed up to our repo.

What am I missing?

Mike.

**Comment by tuves** [comment]
Welcome to the GitHub Community, @mdiehlenator, we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Comment by Dheeraj-147** [comment]
It looks like you're encountering an issue with pushing large files to GitHub, specifically a file that's too large for GitHub’s regular storage. GitHub has a file size limit of 100 MB for individual files, and it seems like the file `ecom_account_116_for_mike_ebay2.csv` exceeds this limit at 162.75 MB.

The error message is pointing you toward Git Large File Storage (Git LFS), a solution for managing large files in Git. Here’s how you can fix the issue and push your code successfully:

---

### **Steps to Fix the Issue Using Git LFS:**

1. **Install Git LFS (Large File Storage)**  
   Git LFS is an extension that allows you to track large files in your repository. To get started:
   - **Install Git LFS** on your machine by following the instructions on their [[Git LFS website](https://git-lfs.github.com/)](https://git-lfs.github.com/).
   - For macOS or Linux, you can install it using a package manager:
     ```bash
     # macOS
     brew install git-lfs

     # Linux
     sudo apt-get install git-lfs
     ```

2. **Initialize Git LFS in Your Repo**  
   After installation, initialize Git LFS in your repository:
   ```bash
   git lfs install
   ```

3. **Track Large Files with Git LFS**  
   You need to tell Git LFS which files or types of files to track. Since the file mentioned in the error (`ecom_account_116_for_mike_ebay2.csv`) is a CSV, we can track all CSV files (or any other file types) with Git LFS. To track the CSV file, run:
   ```bash
   git lfs track "*.csv"
   ```

   This command adds the pattern for `.csv` files to `.gitattributes` in your repository.

4. **Add and Commit the Changes**  
   After tracking the files, add the changes to your local repository:
   ```bash
   git add .gitattributes
   git add <path-to-large-file>  # In your case, 'ecom_account_116_for_mike_ebay2.csv'
   git commit -m "Track large CSV files with Git LFS"
   ```

5. **Push Your Code**  
   Now that the large file is being tracked by Git LFS, you can push your code to GitHub:
   ```bash
   git push --set-upstream origin main
   ```

   Git LFS will handle the large file uploads separately from the regular Git objects, so this should allow you to push your code successfully without hitting the size limit.

---

### **Optional: Removing Large Files from Git History**  
If you’ve already tried to push the file before and it was rejected, Git will still track that large file in the history. You may want to remove the large file from your repository's history to avoid the error in future pushes.

To do this, you can use the `git filter-repo` tool to remove the large file from the history:
1. Install `git-filter-repo`:
   ```bash
   pip install git-filter-repo
   ```

2. Remove the large file from the history:
   ```bash
   git filter-repo --path OLD/orig/ecom_account_116_for_mike_ebay2.csv --invert-paths
   ```

3. After cleaning the history, force push to the repository:
   ```bash
   git push --force origin main
   ```

---

### **Key Takeaways:**
- **Git LFS** is the proper solution for pushing large files to GitHub.
- You need to **track large files** using `git lfs track` and then commit and push them as usual.
- If a large file has already been pushed and rejected, you may need to **clean the file from your Git history** using `git filter-repo`.

Let me know if you need any further clarification or run into any issues along the way!

Best regards,  
Dheeraj B

---

**Comment by BjornFrancke** [comment]
1. **Remove the Large File from the Repository**:
   Since you mentioned that there is nothing in `./OLD/*`, it’s possible that the large file is still being tracked by Git. You can remove it from the staging area and commit the changes. Here’s how to do that:

   ```bash
   git rm --cached OLD/orig/ecom_account_116_for_mike_ebay2.csv
   ```

   This command removes the file from the index (staging area) but keeps it in your working directory.

2. **Commit the Changes**:
   After removing the large file from the index, you need to commit this change:

   ```bash
   git commit -m "Remove large file to comply with GitHub limits"
   ```

3. **Push the Changes**:
   Now that the large file has been removed from the index, you can attempt to push your changes again:

   ```bash
   git push --set-upstream origin main
   ```

**Reply by mdiehlenator** [reply]
Thank you for getting back to me with solid suggestions.  I wasn't aware of the --cached flag for git rm. 

However, I followed your instructions to no avail.  I even tried deleting the file(s) in question from the filesystem and recreating them as zero-byte files.  Here is where I stand now:

**$ git status**
On branch main
nothing to commit, working tree clean

The OLD directory is empty:

**$ ls -la OLD**
total 8
drwxrwxr-x  2 mdiehl mdiehl 4096 Dec 24 12:25 .
drwxrwxr-x 10 mdiehl mdiehl 4096 Dec 24 12:25 ..

But I still can't push to the repo:

**$ git push --set-upstream origin main**
Enumerating objects: 181, done.
Counting objects: 100% (181/181), done.
Delta compression using up to 8 threads
Compressing objects: 100% (174/174), done.
Writing objects: 100% (181/181), 102.92 MiB | 6.13 MiB/s, done.
Total 181 (delta 53), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (53/53), done.
remote: error: Trace: 579146a8a7997b2c25b1478c6344d423dc2ed68bbbc14ff694dc410ad3616a47
remote: error: See https://gh.io/lfs for more information.
remote: error: File OLD/orig/ecom_account_116_for_mike_ebay2.csv is 162.75 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To github.com:webretailgroup/E-Bay_Crawler.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'github.com:webretailgroup/E-Bay_Crawler.git'


So, I continue with your instructions:

**$ git rm --cached OLD/orig/ecom_account_116_for_mike_ebay2.csv**
fatal: pathspec 'OLD/orig/ecom_account_116_for_mike_ebay2.csv' did not match any files

When I try to push, I get the same error message.  (not pasted here)

So, I try to create the file in question:

**$ mkdir OLD/orig**
**$ touch OLD/orig/ecom_account_116_for_mike_ebay2.csv**

So, where do we stand?

**$ git status**
On branch main
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        OLD/

**$ git add OLD**
**$ git status**

On branch main
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   OLD/orig/ecom_account_116_for_mike_ebay2.csv


So, at this point, I can either:

- git restore --staged <file>
- git commit -m 'Hmmm'
- git reset <file>

Either way, I think I'll be back to where I was.  I've been running around in circles on this.  Perhaps I need to blow away my local repo and re-init?

Your suggestions would be most welcome.

Mike.
